### DISCUSSION TITLE
Filtering in Araidne ?

### DISCUSSION THREAD

**Discussion by lexa105** [mirumee/ariadne#732] [Q&A] [answered]
Hello, I'm really new to Ariadne and GraphQL in general.

I was following [this tutorial ](https://www.howtographql.com/graphql-python/7-filtering/)for filtering data published from.

This is my goal I want to achieve >> show only data that contain "custom input" in "published from"
![image](https://user-images.githubusercontent.com/56413595/144449760-5c33d94d-a06c-4423-b4e6-3a59b8387436.png)

But it has problem with my arguments. It always say that "unknown argument" event tho, I already implemented in.

My CODE
`
from typing import Any
from ariadne import QueryType, make_executable_schema, MutationType
from .models import Trailer

query = QueryType()
mutation = MutationType()
@query.field('all_trailers')
def resolve_trailers(*_, published_from = None,):
    
    if published_from:
        filter(
        Trailer(published_from__icontains=published_from)
        )
        return Trailer.objects.filter(filter)

    return Trailer.objects.all()
`
I will be thankful for every response :)

Have a great day :)

**Comment by rafalp** [comment, marked as answer]
To filter data, implement filtering logic in your resolvers. Assuming you are using Django:

```python
def resolve_trailes(*_, published_from=None):
    queryset = Trailer.objects.all()
    if published_from:
        queryset = queryset.filter(published_from__icontains=published_from)
    return queryset
```

**Reply by lexa105** [reply]
Found my problem, I didn't know new arguments has to added into schema. Thanks for helping !

**Comment by rafalp** [comment]
Can you share your GraphQL schema?
